codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ServerDowngrader.java

```diff
@@ -78,9 +80,10 @@ protected void decode(ChannelHandlerContext ctx, Http2StreamFrame frame, List<Ob
                     out.add(full);
                 }
             } else {
-                out.add(HttpConversionUtil.toHttpRequest(id, headersFrame.headers(), validateHeaders));
+                HttpRequest req = HttpConversionUtil.toHttpRequest(id, headersFrame.headers(), validateHeaders);
+                HttpUtil.setTransferEncodingChunked(req, true);
```

Comment:
Hmm... This wouldn't seem to be very normal for content-less requests (like GET). It also removes Content-Length. If there's a Content-Length, it should probably be preferred.